bin/tools.py

```diff
@@ -1074,11 +1111,27 @@ def run_parsed_arguments(args):
                     contest, problems, tmpdir, statement_language, web=True
                 )
                 if not config.args.no_solutions:
-                    success &= latex.build_contest_pdfs(
-                        contest, problems, tmpdir, statement_language, solutions=True
+                    success &= latex.build_contest_pdf(
+                        contest,
+                        problems,
+                        tmpdir,
+                        statement_language,
+                        build_type=latex.PdfType.SOLUTION,
+                    )
+                    success &= latex.build_contest_pdf(
+                        contest,
+                        problems,
+                        tmpdir,
+                        statement_language,
+                        build_type=latex.PdfType.SOLUTION,
+                        web=True,
                     )
-                    success &= latex.build_contest_pdfs(
-                        contest, problems, tmpdir, statement_language, solutions=True, web=True
+                    success &= latex.build_contest_pdf(
```

Comment:
Should this be more granular? It's annoying if things fail because no problem slides are present.

Or maybe better is to never make it an error if problem slides are missing? (But that's kinda tricky when you want slides for all problems?)

Also I checked an doing `bt problem_slides` at the contest level with some `problem-slide.en.tex` missing crashes in latex (because file not found). We should probably detect when there's not a single problem-slide present and then just Log that (without error). When only some files are present, it's fine to error IMO.

At the problem level it gives a nice message `problem-slide.en.tex not found`, so there all is good.

Otherwise LGTM! Thanks for making the PR :)

Reply:
That's a good point! Will reduce some errors to warnings today or tomorrow :smile:

Reply:
I've added some extra checks :slightly_smiling_face: The behaviour at the problem level hasn't changed, this is the new behaviour at the contest level:
- When no `problem-slide.*.tex` are present:
  - `bt problem_slides` will warn for all problems, followed by a failing LaTeX compilation (because the user is explicitly trying to compile them)
  - `bt zip` will log "No problem has problem-slide.*.tex, skipping problem slides"
- When at least one `problem-slide.*.tex` is present:
  - Both `bt problem_slides` and `bt zip` will warn for the problems that do not have it, but will compile the rest

Reply:
That sounds good. Thanks! Merging now.